### PR TITLE
Allow existing models to be updated without uniqueness error

### PR DESCRIPTION
**Pull request by gregpardo**

### CODE REVIEW

**Comment by ahmgeek** [comment]
@gregpardo can you provide a description for this PR please?

**Comment by gregpardo** [comment]
My situation was throwing a not unique validation error even though the field was unique. This was on an update including an update to children of that model. When I investigated I found it was failing validation because that query in this file contained the object itself. There is probably a better way to do this in the query itself. Should the model itself be returned in that query?

**Comment by ahmgeek** [comment]
@gregpardo I am not sure if this is a clean solution or not, I've been able to reproduce the bug, will attach a failing test case to this PR and work on top of it if necessary.

File: lib/uniqueness/model.rb [hunk 1]
```diff
@@ -60,7 +60,8 @@ def uniqueness_validation
           end
           conditions[field] = value
           query = self.class.where(conditions)
-          errors.add(field, 'should be unique') if query.any?
+          results = query.all - [self]
```
Comment (owahab) [inline lib/uniqueness/model.rb]: I'd go for adding to `conditions` instead of this.

Comment (mckelvaney) [reply lib/uniqueness/model.rb]: Hi @owahab, I know that this is an old PR and comment, but would you be able to shed any more light on the use of conditions?

I've attempted to add a condition via the `scope: []` option in the `options` param, but it both appends the scope query to `conditions` and the name of the scope to the `query`, leading to a DB adapter error (Postgres in my situation). The use of `send()` causes a `NoMethodError` if relying purely on a scope.

I've searched around for this pattern to try an obtain better knowledge of how it works (maybe I'm implementing incorrectly), but it has been in vein.  Any guidance would be greatly received, an example added to the README would be even better 😃.

**Comment by kvandake** [comment]
Hello! Any progress?

**Comment by gregpardo** [comment]
Hello all. I've updated my PR with a clearer explanation of what is going on. 

Turns out in the test, 'save' was being called for the update test and not 'save!' which in turn allowed the test to pass but the model was not really updating so the test was flawed. 

After updating the test the test failed which means that the even a model with existing unique field could not ever be updated at all because it was assumed that it already existed in the database. 

So instead of removing self from the query like my original fix, I only allow the validation to occur if the model is a new record (check for unique) or if any of the fields that are unique have been updated and need to be checked again. 

Please let me know if this is acceptable and if it is we can merge in.

**Comment by amrrbakry** [comment]
@gregpardo hey Greg, could you please update the gemfiles so travis check could pass

**Comment by gregpardo** [comment]
@amrrbakry I've reverted any changes to the gemspec but looks like tests are still failing. Any idea?

**Comment by amrrbakry** [comment]
@gregpardo we use the appraisal gem so I think you have to run `$ appraisal update` to update gemfiles and that will make all checks pass

**Comment by gregpardo** [comment]
@amrrbakry I was unable to get tests passing with old ruby and rails versions. I've updated to use

Ruby
2.3.0
2.4.0
2.5.0
2.6.0

Rails
5.0
5.1
5.2
6.0.0beta3

**Comment by rkhater** [comment]
@gregpardo thank you for contribution, Its merged and published to ruby gems
cc @kvandake 

@gregpardo i'll handle the old versions

**Comment by gregpardo** [comment]
@rkhater Realized forgot to update the README to deprecate support for Rails < 5.0 and Ruby < 2.3.0

**Comment by rkhater** [comment]
@gregpardo Actually we didn't deprecate it, I've fixed travis in this PR: https://github.com/eventtus/uniqueness/pull/12

**Pull request merged**